manifests/init.pp

```diff
@@ -76,24 +84,36 @@
   }
 
   # If we are managing the repository, set it up and install the package with a require on the repo
-  if $manage_repo and $facts['os']['family'] in ['Redhat'] {
+  if $manage_repo {
```

Comment:
I think we need to keep the `and` condition here and add `Debian` to the array, so

```
if $manage_repo and $facts['os']['family'] in ['Redhat', 'Debian'] {
```

In the tests it's trying to use a repository for SLES installs, but I don't think they maintain a repo for that platform.  So we need to default to using an archive source for SLES.

Reply:
I would like to keep the condition short and add a `data/os/SLES.yaml` with the value `influxdb::manage_repo: false`

Reply:
I think only validating $manage_repo here is fine, because there's a case statement afterwards for the individual OS families. Personally I would set $manage_repo to false in common.yaml and enable it only for Debian and RedHat OS family.

Reply:
So I misread the failure in the spec tests, my bad.  The actual failure is that `$repo_url` is unset on SLES because it isn't in `common.yaml` and it's a required parameter.  https://github.com/puppetlabs/influxdb/runs/7993767980?check_suite_focus=true#step:10:183

We could make this an `Optional[String]` that defaults to `undef`, or put a default value in `common.yaml`.  I think I like `Optional` more because there isn't one for SLES.

I agree that only checking `$manage_repo` is fine since I was mistaken about the issue.

Reply:
Also, if you wanted to move the defaults for `$manage_repo` to module data that would be fine :+1: